docs/conandata_yml_format.md

```diff
@@ -0,0 +1,220 @@
+
+### sources - use cases
+
+there are several major ways to specify sources.
+
+#### single source
+
+the most straighforward format:
+```
+sources:
+  "1.18.0":
+    sha256: 820688d1e0387ff55194ae20036cbae0fb3c7d11b7c3f46492369723c01df96f
+    url: https://github.com/chriskohlhoff/asio/archive/asio-1-18-0.tar.gz
+```
+
+#### single source with download mirror
+
+sometimes it's a good idea to specify several download mirrors - some hosts are known to be unreliable or banned in certain countries:
+```
+sources:
+  "1.2.11":
+    url: [
+         "https://zlib.net/zlib-1.2.11.tar.gz",
+         "https://downloads.sourceforge.net/project/libpng/zlib/1.2.11/zlib-1.2.11.tar.gz",
+       ]
+    sha256: "c3e5e9fdd5004dcb542feda5ee4f0ff0744628baf8ed2dd5d66f8ca1197cb1a1"
+```
+keep in mind all the mirrors have to provide the exactly same source (e.g. no repackaging), thus using the same hash sum.
+
+#### source code + license
+
+certain projects provide license on its own, and released artifacts do not include it already:
+```
+sources:
+  8.0.0:
+    - url: https://github.com/approvals/ApprovalTests.cpp/releases/download/v.8.0.0/ApprovalTests.v.8.0.0.hpp
+      sha256: e16a97081f8582be951d95a9d53dc611f1f5a84e117a477029890d0b34ae99d6
+    - url: "https://raw.githubusercontent.com/approvals/ApprovalTests.cpp/v.8.0.0/LICENSE"
+      sha256: c71d239df91726fc519c6eb72d318ec65820627232b2f796219e87dcf35d0ab4
+```
+
+#### several source code archives
+
+certain projects may include multiple tarballs as a part of release, [OpenCV](https://opencv.org/) is an example which includes auxillary [contrib](https://github.com/opencv/opencv_contrib) archive:
+```
+sources:
+  "4.5.0":
+    - sha256: dde4bf8d6639a5d3fe34d5515eab4a15669ded609a1d622350c7ff20dace1907
+      url: https://github.com/opencv/opencv/archive/4.5.0.tar.gz
+    - sha256: a65f1f0b98b2c720abbf122c502044d11f427a43212d85d8d2402d7a6339edda
+      url: https://github.com/opencv/opencv_contrib/archive/4.5.0.tar.gz
+```
+
+#### different source code archives per configuration
+
+this is the most advanced and sophisticated use-case, fortunately, rarely encountered in the wild. project may provide different sources for different platforms for awkward reasons, it could be expressed as:
+
+```
+sources:
+  "0066":
+    "Macos":
+      "apple-clang":
+        "x86":
+          - url: "https://naif.jpl.nasa.gov/pub/naif/misc/toolkit_N0066/C/MacIntel_OSX_AppleC_32bit/packages/cspice.tar.Z"
+            sha256: "9a4b5f674ea76821c43aa9140829da4091de646ef3ce40fd5be1d09d7c37b6b3"
+        "x86_64":
+          - url: "https://naif.jpl.nasa.gov/pub/naif/misc/toolkit_N0066/C/MacIntel_OSX_AppleC_64bit/packages/cspice.tar.Z"
+            sha256: "f5d48c4b0d558c5d71e8bf6fcdf135b0943210c1ff91f8191dfc447419a6b12e"
+```
+
+this approach requires a special code within [source](https://docs.conan.io/en/latest/reference/conanfile/methods.html#source) method to handle.
```

Comment:
If sources are different per configuration, they should be retrieved in the `build()` method. Source folder is shared for all the configurations, it is not a problem for ConanCenter (always uses a clean cache), but better to avoid.

Reply:
yes, that's right. still, it's the logic of the build method, not the part of the `conandata.yml` format.